packages/facility-server/__tests__/apiv1/SurveyResponse.test.js

```diff
@@ -196,6 +262,152 @@ describe('SurveyResponse', () => {
     });
   });
 
+  describe('complexChartInstance', () => {
+    it('should update existing answers', async () => {
+      const { answer, response } = await setupComplexChartSurvey();
+      const oldValue = answer.body;
+      const newValue = 'Updated answer';
+
+      const result = await app.put(`/api/surveyResponse/complexChartInstance/${response.id}`).send({
+        answers: {
+          [answer.dataElementId]: newValue,
+        },
+      });
+      await answer.reload();
+
+      expect(result).toHaveSucceeded();
+      expect(answer.body).toEqual(newValue);
+      expect(answer.body).not.toEqual(oldValue);
+    });
+
+    it('should create new answers for data elements not previously answered', async () => {
+      const { ProgramDataElement, SurveyScreenComponent } = models;
+      const { response } = await setupComplexChartSurvey();
+      const newDataElement = await ProgramDataElement.create({
+        ...fake(ProgramDataElement),
+        type: PROGRAM_DATA_ELEMENT_TYPES.TEXT,
+      });
+      await SurveyScreenComponent.create({
+        ...fake(SurveyScreenComponent),
+        dataElementId: newDataElement.id,
+        surveyId: response.surveyId,
+        calculation: '',
+      });
+      const newValue = 'New answer value';
+
+      const result = await app.put(`/api/surveyResponse/complexChartInstance/${response.id}`).send({
+        answers: {
+          [newDataElement.id]: newValue,
+        },
+      });
+      const newAnswer = await models.SurveyResponseAnswer.findOne({
+        where: {
+          responseId: response.id,
+          dataElementId: newDataElement.id,
+        },
+      });
+
+      expect(result).toHaveSucceeded();
+      expect(newAnswer).toBeTruthy();
+      expect(newAnswer.body).toEqual(newValue);
+    });
+
+    it('should ignore null values in answers', async () => {
+      const { answer, response } = await setupComplexChartSurvey();
+      const originalValue = answer.body;
+
+      const result = await app.put(`/api/surveyResponse/complexChartInstance/${response.id}`).send({
+        answers: {
+          [answer.dataElementId]: null,
+        },
+      });
+      await answer.reload();
+
+      expect(result).toHaveSucceeded();
+      expect(answer.body).toEqual(originalValue);
+    });
+
+    it('should update multiple answers in a single request', async () => {
```

Comment:
Might be nice to have a new answer or two created in the same hit as the multiple updates?